rasa/nlu/emulators/luis.py

```diff
@@ -1,50 +1,95 @@
+    https://docs.microsoft.com/en-us/azure/cognitive-services/LUIS/luis-concept-data-extraction?tabs=V3
+    """
 
     def _top_intent(self, data) -> Optional[Dict[Text, Any]]:
-        if data.get("intent"):
+        intent = data.get(INTENT)
+
+        if not intent:
+            return None
+
+        return {
+            "intent": intent[INTENT_NAME_KEY],
+            "score": intent[PREDICTED_CONFIDENCE_KEY],
+        }
+
+    def _intents(self, data) -> Dict[Text, Any]:
+        if data.get(INTENT_RANKING_KEY):
             return {
-                "intent": data["intent"]["name"],
-                "score": data["intent"]["confidence"],
+                el[INTENT_NAME_KEY]: {"score": el[PREDICTED_CONFIDENCE_KEY]}
+                for el in data[INTENT_RANKING_KEY]
             }
-        else:
-            return None
 
-    def _ranking(self, data) -> List[Dict[Text, Any]]:
-        if data.get("intent_ranking"):
-            return [
-                {"intent": el["name"], "score": el["confidence"]}
-                for el in data["intent_ranking"]
-            ]
-        else:
-            top = self._top_intent(data)
-            return [top] if top else []
+        top = self._top_intent(data)
+        if not top:
+            return {}
 
-    def normalise_response_json(self, data: Dict[Text, Any]) -> Dict[Text, Any]:
-        """Transform data to luis.ai format."""
+        return {top["intent"]: {"score": top["score"]}}
 
-        top_intent = self._top_intent(data)
-        ranking = self._ranking(data)
-        return {
-            "query": data["text"],
-            "topScoringIntent": top_intent,
-            "intents": ranking,
-            "entities": [
+    def _entities(self, data) -> Optional[Dict[Text, Any]]:
+        if ENTITIES not in data:
+            return {}
+
+        entities = {"$instance": {}}
+        for e in data[ENTITIES]:
+            # LUIS API v3 uses entity roles instead of entity names
+            # (it's possible because its roles are unique):
+            # https://docs.microsoft.com/en-us/azure/cognitive-services/LUIS/luis-migration-api-v3#entity-role-name-instead-of-entity-name
+            key = e.get(ENTITY_ATTRIBUTE_ROLE, e[ENTITY_ATTRIBUTE_VALUE])
+
+            entities.update({key: [e[ENTITY_ATTRIBUTE_VALUE]]})
+            entities["$instance"].update(
                 {
-                    "entity": e["value"],
-                    "type": e["entity"],
-                    "startIndex": e.get("start"),
-                    "endIndex": (e["end"] - 1) if "end" in e else None,
-                    "score": e.get("confidence"),
+                    "role": e.get(ENTITY_ATTRIBUTE_ROLE),
+                    "type": e[ENTITY_ATTRIBUTE_TYPE],
+                    "text": e[ENTITY_ATTRIBUTE_VALUE],
+                    "startIndex": e.get(ENTITY_ATTRIBUTE_START),
+                    "length": len(e[ENTITY_ATTRIBUTE_VALUE]),
```

Comment:
`ENTITY_ATTRIBUTE_VALUE` could be something mapped. Should we rather use something like `end_index - start_index`?